Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,639 @@
+# coding=utf-8
+"""
+Written by Egbert Bouman, Mihai Capotă, Elric Milon, and Ardhi Putra Pratama H
+Supported boosting sources
+"""
+import HTMLParser
+import glob
+import logging
+import os
+import random
+import re
+import time
+import urllib
+from binascii import hexlify
+from hashlib import sha1
+
+import libtorrent as lt
+from twisted.internet import reactor
+from twisted.internet.task import LoopingCall
+from twisted.web.client import Agent, readBody
+from twisted.web.http_headers import Headers
+
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.simpledefs import NTFY_INSERT, NTFY_TORRENTS, NTFY_UPDATE, NTFY_CHANNELCAST, NTFY_VOTECAST
+from Tribler.Core.version import version_id
+from Tribler.Main.Utility.GuiDBTuples import Torrent, CollectedTorrent, RemoteTorrent, NotCollectedTorrent, Channel, \
+    ChannelTorrent
+from Tribler.community.allchannel.community import AllChannelCommunity
+from Tribler.community.channel.community import ChannelCommunity
+from Tribler.dispersy.exception import CommunityNotFoundException
+from Tribler.dispersy.taskmanager import TaskManager
+from Tribler.dispersy.util import call_on_reactor_thread
+
+def ent2chr(input_str):
+    """
+    Function to unescape literal string in XML to symbols
+    source : http://www.gossamer-threads.com/lists/python/python/177423
+    """
+    code = input_str.group(1)
+    code_int = int(code) if code.isdigit() else int(code[1:], 16)
+    return chr(code_int) if code_int < 256 else '?'
+
+
+class TorrentManagerCM(TaskManager):
+    """
+    *Temporary* class to handle load torrent.
+
+    Adapted from TorrentManager in SearchGridManager
+    """
+    __single = None
+
+    def __init__(self, session):
+        super(TorrentManagerCM, self).__init__()
+        TorrentManagerCM.__single = self
+
+        self.session = session
+        self.torrent_db = self.session.open_dbhandler(NTFY_TORRENTS)
+        self.channelcast_db = self.session.open_dbhandler(NTFY_CHANNELCAST)
+        self.votecastdb = self.session.open_dbhandler(NTFY_VOTECAST)
+
+        self.dslist = []
+
+    # TODO(ardhi) : temporary function until GUI and core code are separated
+    def load_torrent(self, torrent, callback=None):
```

Comment:
It appears you need the `Torrent` object from `GUIDBTuples`. Is it absolutely necessary to construct a `Torrent` object? If you only need some specific attributes from a torrent (i.e, name, info hash, files), maybe we can avoid using the `Torrent` object (which will either be removed or moved to the `Core` after the big wx refactor).